yellowpages/yellowpages-war/src/main/java/org/silverpeas/components/yellowpages/control/YellowpagesSessionController.java

```diff
@@ -439,26 +430,15 @@ public void deleteContactFathers(String contactId) {
     resetCurrentFullCompleteUsers();
   }
 
-  public Collection<UserContact> getContacts(Collection<String> targetIds) {
-    return getYellowpagesService().getContacts(targetIds, getComponentId());
-  }
-
-  /**
-   * methods for Users
-   * @return
-   */
-  public UserDetail[] getUserList() {
-    return getOrganisationController().getAllUsers();
-  }
-
   /**
    * get others instances of yellowpages
    * @return
    */
   public CompoSpace[] getYellowPagesInstances() {
-    CompoSpace[] compoSpaces =
-        getOrganisationController().getCompoForUser(getUserId(), "yellowpages");
-    return compoSpaces;
+    if (StringUtil.getBooleanValue(getComponentParameterValue("directAccess"))) {
+      return getOrganisationController().getCompoForUser(getUserId(), "yellowpages");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "yellowpages" 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)